sdks/wasm/framework/src/WebAssembly.Bindings/Runtime.cs

```diff
@@ -76,23 +117,56 @@ public static JSObject NewJSObject (JSObject js_func_ptr = null, params object [
 			return res as JSObject;
 		}
 
-		static int BindJSObject (int js_id)
+		static int BindJSObject (int js_id, string type)
 		{
 			JSObject obj;
-			if (!bound_objects.TryGetValue (js_id, out obj))
-				bound_objects [js_id] = obj = new JSObject (js_id);
+			if (!bound_objects.TryGetValue (js_id, out obj)) {
+				if (js_clr_mapping.TryGetValue (type, out Type mappedType)) {
+					return BindJSType (js_id, mappedType);
+				} else {
+					bound_objects [js_id] = obj = new JSObject ((IntPtr)js_id);
+				}
+			}
 
 			return (int)(IntPtr)obj.Handle;
 		}
 
+		static int BindCoreCLRObject (int js_id, int gcHandle)
+		{
+			//Console.WriteLine ($"Registering CLR Object {js_id} with handle {gcHandle}");
+			GCHandle h = (GCHandle)(IntPtr)gcHandle;
+			JSObject obj = (JSObject)h.Target;
+
+			if (bound_objects.ContainsKey (js_id))
+				obj = bound_objects [js_id];
+			else
+				bound_objects [js_id] = obj;
+
+			return (int)(IntPtr)obj.Handle;
+		}
+
+		static int BindJSType (int js_id, Type type)
+		{
+			//Console.WriteLine ($"BindJSType: {js_id} of {type} intptr: {(IntPtr)js_id}");
+			JSObject obj;
+			if (!bound_objects.TryGetValue (js_id, out obj)) {
+				var jsobjectnew = type.GetConstructor (BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.ExactBinding,
+			    		null, new Type [] { typeof (IntPtr) }, null);
+				bound_objects [js_id] = obj = (JSObject)jsobjectnew.Invoke (new object [] { (IntPtr)js_id });
+			}
+			return (int)(IntPtr)obj.Handle;
+		}
+
 		static int UnBindJSObject (int js_id)
 		{
-			if (bound_objects.TryGetValue (js_id, out var obj)) {
+			if (bound_objects.ContainsKey (js_id)) {
+				var obj = bound_objects [js_id];
```

Comment:
Why was this changed to the less efficient `ContainsKey(…) + this[…]` pattern?

Reply:
My bad after conflict resolution.  Thanks for the catch.

Reply:
Should be fixed